src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```
@@ -711,27 +692,54 @@ public Transformer(IHostEnvironment env, ModelLoadContext ctx)
                 _host.CheckValue(ctx, nameof(ctx));
 
                 ctx.CheckAtModel(GetVersionInfo());
-                int n = ctx.Reader.ReadInt32();
-
-                ctx.LoadModel<ILegacyDataLoader, SignatureLoadDataLoader>(env, out var loader, "Loader", new MultiFileSource(null));
 
-                IDataView data = loader;
-                for (int i = 0; i < n; i++)
+                if (ctx.Header.ModelVerReadable == VerIDataTransform)
                 {
-                    var dirName = string.Format(TransformDirTemplate, i);
-                    ctx.LoadModel<IDataTransform, SignatureLoadDataTransform>(env, out var xf, dirName, data);
-                    data = xf;
-                }
+                    int n = ctx.Reader.ReadInt32();
+                    _chain = new TransformerChain<ITransformer>();
+                    ctx.LoadModel<ILegacyDataLoader, SignatureLoadDataLoader>(env, out var loader, "Loader", new MultiFileSource(null));
+                    IDataView data = loader;
+                    for (int i = 0; i < n; i++)
+                    {
+                        var dirName = string.Format(TransformDirTemplate, i);
+                        ITransformer transformer;
+                        // Try to load as an ITransformer.
+                        try
+                        {
+                            ctx.LoadModelOrNull<ITransformer, SignatureLoadModel>(env, out transformer, dirName);
+                        }
+                        catch
```

Are we OK with eating all exceptions here?